packages/react-router-native/index.tsx

```diff
@@ -158,35 +160,38 @@ const URLEventType = 'url';
 /**
  * Enables support for the hardware back button on Android.
  */
-export function useHardwareBackButton() {
+export function useAndroidBackButton() {
+  const index = useIndex();
+  const navigate = useNavigate();
+
+  const handleHardwardBackPress = React.useCallback(() => {
+    if (index === 0) {
+      return false; // home screen
+    } else {
+      navigate(-1);
+      return true;
+    }
+  }, [index, navigate]);
+
+  const handleHardwardBackPressRef = React.useRef(handleHardwardBackPress);
+
   React.useEffect(() => {
-    function handleHardwardBackPress() {
-      return undefined;
-      // TODO: The implementation will be something like this
-      // if (history.index === 0) {
-      //   return false; // home screen
-      // } else {
-      //   history.back();
-      //   return true;
-      // }
+    handleHardwardBackPressRef.current = handleHardwardBackPress;
+  }, [handleHardwardBackPress]);
+
+  React.useEffect(() => {
+    function eventHandler() {
+      return handleHardwardBackPressRef.current();
     }
 
-    BackHandler.addEventListener(
-      HardwareBackPressEventType,
-      handleHardwardBackPress
-    );
+    BackHandler.addEventListener(HardwareBackPressEventType, eventHandler);
 
     return () => {
-      BackHandler.removeEventListener(
-        HardwareBackPressEventType,
-        handleHardwardBackPress
-      );
+      BackHandler.removeEventListener(HardwareBackPressEventType, eventHandler);
     };
   }, []);
 }
 
-export { useHardwareBackButton as useAndroidBackButton };
```

Comment:
I'd prefer to keep the double exports. `useAndroidBackButton` is essentially just an alias for `useHardwareBackButton`.

Reply:
I changed this last minute because I realized the double export does not appear in the generated build (`useHardwareBackButton` was there but `useAndroidBackButton` was not). 🤷 I will have a second look what might be causing this.

Reply:
It seems to be a `rollup` feature (issue?) to not allow multiple names for one export. But changing `export { useHardwareBackButton as useAndroidBackButton };` to `export const useAndroidBackButton = useHardwareBackButton;` works. I also added a test to check that the re-export is available.